connectors/sources/salesforce.py

```diff
@@ -80,58 +110,66 @@ async def close(self):
         retries=RETRIES,
         interval=RETRY_INTERVAL,
     )
-    async def get_token(self, enable_refresh=True):
-        # TODO add refresh method
-        self._logger.debug("Fetching Salesforce token...")
-        resp = await self.session.post(
-            f"{self.base_url}{TOKEN_ENDPOINT}", data=self.token_payload
-        )
-        resp_json = await resp.json()
-        self.token = resp_json["access_token"]
-        self.token_issued_at = resp_json["issued_at"]
-        self.token_refresh_enabled = enable_refresh
-        self._logger.debug("Salesforce token retrieved.")
+    async def get_token(self):
+        if self._fetching_token is True:
+            return
+
+        self._fetching_token = True
+        response_body = None
+        try:
+            response = await self._post(
+                f"{self.base_url}{TOKEN_ENDPOINT}", data=self.token_payload
+            )
+            response_body = await response.json()
+            response.raise_for_status()
+            self.token = response_body["access_token"]
+        except ClientResponseError as e:
+            if 400 <= e.status < 500:
+                # 400s have detailed error messages in body
+                error_resp = self._handle_token_response_body_error(response_body)
```

Comment:
Is `e.message` different than response_body?

Reply:
Yeah they are different, for example if the credentials are invalid it throws a 400 with the message a `bad_request`. The actual cause (`invalid_client`) is then inside the response body.